src/psyclone/psyir/nodes/routine.py

```diff
@@ -174,27 +171,43 @@ def name(self, new_name):
         :param str new_name: new name for the Routine.
 
         :raises TypeError: if new_name is not a string.
+        :raises KeyError: if there already is a different named symbol with
+            the 'own_routine_tag' in the symbol_table.
 
         '''
         if not isinstance(new_name, six.string_types):
             raise TypeError("Routine name must be a str but got "
                             "'{0}'".format(type(new_name).__name__))
+        # TODO #1200 The name is duplicated in the _name attribute and the
+        # symbol.name that there is in the local symbol table. This setter
+        # updates both but note that a better solution is needed because
+        # renaming the symbol_table symbol alone would make it inconsistent.
         if not self._name:
+            # If the 'own_routine_symbol' tag already exist (if a specific
+            # symbol table has been provided in the creator) check that is
```

Comment:
I removed the parenthesis since it made this confusing, but the problem is not if the SymbolTable is new or pre-existing, is that if it is preexisting it may already have an 'own_routine_symbol' tag and this will collide with the one we are about to assign.